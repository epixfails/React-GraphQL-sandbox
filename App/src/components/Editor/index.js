export { editor, setEditor, changeNote } from './ducks';
export Editor from './Editor';
