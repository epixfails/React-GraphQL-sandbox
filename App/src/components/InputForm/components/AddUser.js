import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { addUser } from '../../../ducks';

const Wrapper = styled.div`
  width: 400px;
  margin: auto;
`;

const Input = styled.input`
  display: block;
  width: 100%;
  padding: 5px 10px;
  border: 1px #252525 solid;
  margin: 10px 0;
`;

const Button = styled.button`
  width: 250px;
  line-height: 24px;
  background: #4a76a8;
`;

class AddUser extends Component {
  getNameRef = input => {
    this.nameInput = input;
  };
  getAddressRef = input => {
    this.addressInput = input;
  };

  handleAdd() {
    this.props.addUser({
      name: this.nameInput.value.trim() || 'not specified',
      address: this.addressInput.value.trim() || 'not specified',
    });
  }

  render() {
    return (
      <Wrapper>
        <Input innerRef={this.getNameRef} placeholder="Name" />
        <Input innerRef={this.getAddressRef} placeholder="Address" />
        <Button onClick={() => this.handleAdd()}>Add to list</Button>
      </Wrapper>
    );
  }
}

const mapStateToProps = state => ({
  users: state.users,
});

const mapDispatchToProps = {
  addUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(AddUser);
