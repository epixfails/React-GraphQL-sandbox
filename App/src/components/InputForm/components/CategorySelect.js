import React from 'react';
import styled from 'styled-components';
import { categoryColors } from '~/components/common/styles';

const Wrapper = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin-bottom: 20px;
  flex-flow: row wrap;
`;

const Item = styled.div`
  border-radius: 5px;
  padding: 3px 5px;
  margin: 5px;
  width: 45%;
  cursor: pointer;
  box-sizing: border-box;
  text-align: center;
  transition: 0.3s all;

  ${({ active, item }) =>
    active
      ? `background: ${categoryColors[item]}; color: #fff;`
      : 'background: #fff; color: #90caf9;'};

  & :hover {
    ${({ item }) => `background: ${categoryColors[item]}`};
    color: #fff;
  }
`;

export const CategorySelect = props => {
  const { categories, selected, handleChange } = props;
  return (
    <Wrapper>
      {categories.map((item, i) => (
        <Item
          key={i}
          item={item}
          active={selected === item}
          onClick={() => handleChange(item)}
        >
          {item}
        </Item>
      ))}
    </Wrapper>
  );
};
