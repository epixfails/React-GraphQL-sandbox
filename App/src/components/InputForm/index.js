export AddUser from './components/AddUser';
export addUserSaga from './saga';
