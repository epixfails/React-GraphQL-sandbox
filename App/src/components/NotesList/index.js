export NotesList from './components/NotesList';
export getNotesSaga from './saga';
