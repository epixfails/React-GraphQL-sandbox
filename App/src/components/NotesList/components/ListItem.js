import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { getDateFormatted } from '~/utils';
import { deleteNote, updateNote } from '~/ducks/notes';
import { setEditor } from '~/ducks/editor';
import { setFilter } from '~/ducks/filter';
import { categories } from '~/components/common/categories';
import { Editor } from '../../Editor';

const Wrapper = styled.div`
  padding: 15px 10px;
  cursor: pointer;
  transition: all 0.3s;

  &:hover {
    background: #90caf9;
  }
`;

const Label = styled.div`
  width: 20px;
  height: 20px;
  margin-right: 10px;
  cursor: pointer;
  ${({ category }) => `background: ${categories[category]}`};
`;

const Text = styled.span`
  display: block;
  height: 24px;
  line-height: 24px;
  width: 40%;
  overflow: hidden;
  word-wrap: break-word;
  text-overflow: ellipsis;

  ${({ content }) => (content ? 'font-size: 14px' : 'font-size: 18px')};
`;

const DateEdited = styled.p`
  color: #747474;
  margin: 0 0 5px 0;
  font-size: 10px;
  text-align: left;
`;

const NoteContent = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
`;

const Content = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

const ButtonAction = styled.a`
  display: block;
  padding: 5px 10px;
  cursor: pointer;
  opacity: 0;
  color: #4a148c;
  font-size: 14px;
  transition: all 0.3s;

  & :hover {
    color: #fff;
  }

  ${Wrapper}:hover & {
    opacity: 1;
  }
`;

const ListItem = props => {
  const { note, noteInEditor } = props;
  const isEdited = noteInEditor.id === note.id;
  return (
    <div>
      {isEdited ? (
        <Editor />
      ) : (
        <Wrapper>
          <DateEdited>
            Created: {getDateFormatted(note.date_updated)}
          </DateEdited>
          <Content>
            <Label
              category={note.category}
              onClick={() => props.setFilter(note.category)}
            />
            <NoteContent onClick={() => props.setEditor(note)}>
              <Text>{note.title}</Text>
              <Text content="true">{note.content}</Text>
              <ButtonAction onClick={() => props.deleteNote(note.id)}>
                remove
              </ButtonAction>
            </NoteContent>
          </Content>
        </Wrapper>
      )}
    </div>
  );
};

const mapStateToProps = state => ({
  noteInEditor: state.editor,
});

const mapDispatchToProps = {
  deleteNote,
  updateNote,
  setEditor,
  setFilter,
};

export default connect(mapStateToProps, mapDispatchToProps)(ListItem);
