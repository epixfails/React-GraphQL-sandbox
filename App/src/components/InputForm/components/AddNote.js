import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { addNote } from '~/ducks/notes';
import { categories } from '~/components/common/categories';
import { CategorySelect } from './CategorySelect';

const Wrapper = styled.form`
  display: flex;
  flex-flow: column;
  width: 50%;
  height: 100%;
  padding: 15px 5px;
  background: rgba(144, 202, 249, 0.2);
  box-shadow: 4px 0px 21px -2px rgba(0, 0, 0, 0.3);
`;

const Input = styled.input`
  display: block;
  box-sizing: border-box;
  width: 100%;
  margin-bottom: 15px;
  padding: 5px 10px;
  font-size: 18px;
  border: 1px #cae7fe solid;

  &:focus {
    border-color: #9dd3fe;
  }
`;

const InputContent = styled.textarea`
  display: block;
  box-sizing: border-box;
  width: 100%;
  margin-bottom: 15px;
  padding: 5px 10px;
  font-size: 18px;
  border: 1px #cae7fe solid;
  resize: none;
  outline: none;

  &:focus {
    border-color: #9dd3fe;
  }
`;

const Button = styled.button`
  line-height: 24px;
  font-size: 18px;
  padding: 10px;
  background: #90caf9;
  transition: all 0.3s;
  cursor: pointer;
  border: none;
  color: #fff;

  & :hover {
    color: #fff;
    background: #3ea6fb;
  }
`;

class AddNote extends Component {
  state = {
    category: 'other',
  };
  getNameRef = input => {
    this.nameInput = input;
  };
  getAddressRef = input => {
    this.addressInput = input;
  };

  getFormRef = form => {
    this.form = form;
  };

  getCategoryRef = form => {
    this.categoryInput = form;
  };

  handleCategoryChange = category => {
    this.setState({
      category,
    });
  };

  handleKeyUp = e => {
    if (e.keyCode === 13) {
      this.handleAdd(e);
    }
  };

  handleAdd(e) {
    e.preventDefault();
    this.props.addNote({
      title: this.nameInput.value.trim() || 'not specified',
      content: this.addressInput.value.trim() || 'not specified',
      category: this.state.category,
    });
    this.form.reset();
  }

  render() {
    return (
      <Wrapper innerRef={this.getFormRef} onSubmit={e => this.handleAdd(e)}>
        <Input
          innerRef={this.getNameRef}
          placeholder="Title"
          onKeyUp={this.handleKeyUp}
        />
        <InputContent
          innerRef={this.getAddressRef}
          onKeyUp={this.handleKeyUp}
          placeholder="Content"
          rows="5"
        />
        <CategorySelect
          categories={Object.keys(categories)}
          selected={this.state.category}
          handleChange={this.handleCategoryChange}
        />
        <Button>Add note</Button>
      </Wrapper>
    );
  }
}

const mapStateToProps = state => ({
  notes: state.notes,
});

const mapDispatchToProps = {
  addNote,
};

export default connect(mapStateToProps, mapDispatchToProps)(AddNote);
