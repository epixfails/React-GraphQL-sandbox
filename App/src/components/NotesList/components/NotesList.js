import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import ListItem from './ListItem';

const Wrapper = styled.div`
  margin: 40px auto;
`;

const NotesList = props => {
  const { notesArray } = props;
  return (
    <Wrapper>
      {notesArray.length > 0 &&
        notesArray.map(note => <ListItem note={note} key={note.id} />)}
    </Wrapper>
  );
};

const mapStateToProps = state => ({
  notesArray: state.notes.list || [],
});

export default connect(mapStateToProps)(NotesList);
