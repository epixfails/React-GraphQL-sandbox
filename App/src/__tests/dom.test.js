import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CategorySelect } from '~/components/InputForm/components/CategorySelect';

configure({ adapter: new Adapter() });

describe('<CategorySelect />', () => {
  it('should render CategorySelect', () => {
    const renderedComponent = shallow(
      <CategorySelect categories={[1, 2, 3]} />,
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
