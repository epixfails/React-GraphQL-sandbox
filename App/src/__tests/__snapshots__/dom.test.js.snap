// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategorySelect /> should render CategorySelect 1`] = `
<CategorySelect__Wrapper>
  <CategorySelect__Item
    active={false}
    item={1}
    key="0"
    onClick={[Function]}
  >
    1
  </CategorySelect__Item>
  <CategorySelect__Item
    active={false}
    item={2}
    key="1"
    onClick={[Function]}
  >
    2
  </CategorySelect__Item>
  <CategorySelect__Item
    active={false}
    item={3}
    key="2"
    onClick={[Function]}
  >
    3
  </CategorySelect__Item>
</CategorySelect__Wrapper>
`;
